Content.Client/Mining/MiningOverlaySystem.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared.Mining.Components;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Player;
+
+namespace Content.Client.Mining;
+
+/// <summary>
+/// This handles the lifetime of the <see cref="MiningOverlay"/> for a given entity.
+/// </summary>
+public sealed class MiningOverlaySystem : EntitySystem
+{
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly IOverlayManager _overlayMan = default!;
+
+    private MiningOverlay _overlay = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MiningScannerViewerComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<MiningScannerViewerComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<MiningScannerViewerComponent, LocalPlayerAttachedEvent>(OnPlayerAttached);
+        SubscribeLocalEvent<MiningScannerViewerComponent, LocalPlayerDetachedEvent>(OnPlayerDetached);
+
+        _overlay = new();
+    }
+
+    private void OnPlayerAttached(Entity<MiningScannerViewerComponent> ent, ref LocalPlayerAttachedEvent args)
+    {
+        _overlayMan.AddOverlay(_overlay);
+    }
+
+    private void OnPlayerDetached(Entity<MiningScannerViewerComponent> ent, ref LocalPlayerDetachedEvent args)
+    {
+        _overlay.RequestScreenTexture = false;
```

Comment:
```suggestion
```
This part is not needed since the Overlay gets detached anyways, so set it using `public override bool RequestScreenTexture => true;` in the overlay instead. I do it this way in the Flash overlay since there we only need the first frame and deactivate the request afterwards to safe performance.